src/middleware/jsonld.ts

```diff
@@ -1,26 +1,39 @@
+import ParserJsonld from '@rdfjs/parser-jsonld';
+import SerializerJsonld from '@rdfjs/serializer-jsonld-ext';
 import { format as formatContentType } from 'content-type';
-import jsonld from 'jsonld';
-import { Context as JsonLdContext } from 'jsonld/jsonld-spec';
+import { constants } from 'http2';
+import { Context } from 'jsonld/jsonld-spec';
 import {
-  Context, Middleware, Next, Response,
+  DefaultState, Middleware, Next, Response,
 } from 'koa';
+import pEvent from 'p-event';
+import { fromStream, toStream } from 'rdf-dataset-ext';
+import { DatasetContext } from './dataset';
 
-const isJsonLd = (response: Response): boolean => response.is('jsonld') && typeof response.body === 'object';
+const responseHasContent = (response: Response): boolean => (
+  response.body || response.status === constants.HTTP_STATUS_NO_CONTENT
+);
+
+export default (context: Context = {}): Middleware<DefaultState, DatasetContext> => {
+  const contentType = {
+    type: 'application/ld+json',
+    parameters: { profile: 'http://www.w3.org/ns/json-ld#compacted' },
+  };
+  const parser = new ParserJsonld();
+  const serializer = new SerializerJsonld({ compact: true, context });
+
+  return async ({ request, response }: DatasetContext, next: Next): Promise<void> => {
+    if (request.is('jsonld')) {
+      request.dataset = await fromStream(request.dataset, parser.import(request.req));
```

Comment:
weird signature where the dataset has to be passed in and then is reassigned. Probably can't do much about it

Reply:
Probably don't need the reassignment...